lib/tooling/clang-tidy-tool.ts

```diff
@@ -41,14 +42,25 @@ export class ClangTidyTool extends BaseTool {
         this.addOptionsToToolArgs = false;
     }
 
-    async runTool(compilationInfo, inputFilepath, args, stdin, supportedLibraries) {
+    override async runTool(
+        compilationInfo: Record<any, any>,
```

Comment:
I think this is a `CompilationInfo`. We can defer updating this to another PR if needed (I see `Record<any, any>` is used throughout the lib/tooling code)

Reply:
Yea, I think I checked out CompilationInfo but it had extra fields that wasn't used in the tooling code so chose to stick with the current `Record<any, any>`